### ISSUE TITLE
--check-proofs fails but --check-unsat-cores --check-proofs succeeds

### ISSUE THREAD

**Issue by 4tXJ7f** [CVC4/CVC4#1954]
For one of our regression tests, `--check-proofs` alone fails while the combination `--check-unsat-cores --check-proofs` succeeds (see output below). The issue might be that there is a condition somewhere that only checks for unsat cores being enabled instead of also checking if proof generation is enabled.

```
$ builds/bin/cvc4 --check-proofs test/regress/regress0/bv/core/slice-15.smt
/tmp/cvc4_USj3xA, line 1923, column 163:
The type expected for a symbol does not match the symbol's type.
1. The symbol: bb.l66
2. The expected type: (th_holds (not (and (iff (bitof BVSKOLEM$$_1 0) (bitof BVSKOLEM$$_1 1)) (and (iff (bitof BVSKOLEM$$_1 1) (bitof BVSKOLEM$$_1 2)) (and (iff (bitof BVSKOLEM$$_1 2) (bitof BVSKOLEM$$_1 3)) (and (iff (bitof BVSKOLEM$$_1 3) (bitof BVSKOLEM$$_1 4)) (and (iff (bitof BVSKOLEM$$_1 4) (bitof BVSKOLEM$$_1 5)) (and (iff (bitof BVSKOLEM$$_1 5) (bitof BVSKOLEM$$_1 6)) (and (iff (bitof BVSKOLEM$$_1 6) (bitof BVSKOLEM$$_1 7)) (and (iff (bitof BVSKOLEM$$_1 7) (bitof BVSKOLEM$$_1 8)) (and (iff (bitof BVSKOLEM$$_1 8) (bitof BVSKOLEM$$_1 9)) (and (iff (bitof BVSKOLEM$$_1 9) (bitof BVSKOLEM$$_1 10)) (and (iff (bitof BVSKOLEM$$_1 10) (bitof BVSKOLEM$$_1 11)) (and (iff (bitof BVSKOLEM$$_1 11) (bitof BVSKOLEM$$_1 12)) (and (iff (bitof BVSKOLEM$$_1 12) (bitof BVSKOLEM$$_1 13)) (and (iff (bitof BVSKOLEM$$_1 13) (bitof BVSKOLEM$$_1 14)) (iff (bitof BVSKOLEM$$_1 14) true)))))))))))))))))
3. The symbol's type: (th_holds (not (and (iff (bitof BVSKOLEM$$_1 1) (bitof BVSKOLEM$$_1 0)) (and (iff (bitof BVSKOLEM$$_1 2) (bitof BVSKOLEM$$_1 1)) (and (iff (bitof BVSKOLEM$$_1 3) (bitof BVSKOLEM$$_1 2)) (and (iff (bitof BVSKOLEM$$_1 4) (bitof BVSKOLEM$$_1 3)) (and (iff (bitof BVSKOLEM$$_1 5) (bitof BVSKOLEM$$_1 4)) (and (iff (bitof BVSKOLEM$$_1 6) (bitof BVSKOLEM$$_1 5)) (and (iff (bitof BVSKOLEM$$_1 7) (bitof BVSKOLEM$$_1 6)) (and (iff (bitof BVSKOLEM$$_1 8) (bitof BVSKOLEM$$_1 7)) (and (iff (bitof BVSKOLEM$$_1 9) (bitof BVSKOLEM$$_1 8)) (and (iff (bitof BVSKOLEM$$_1 10) (bitof BVSKOLEM$$_1 9)) (and (iff (bitof BVSKOLEM$$_1 11) (bitof BVSKOLEM$$_1 10)) (and (iff (bitof BVSKOLEM$$_1 12) (bitof BVSKOLEM$$_1 11)) (and (iff (bitof BVSKOLEM$$_1 13) (bitof BVSKOLEM$$_1 12)) (and (iff (bitof BVSKOLEM$$_1 14) (bitof BVSKOLEM$$_1 13)) (iff true (bitof BVSKOLEM$$_1 14))))))))))))))))))
$ builds/bin/cvc4 --check-unsat-cores --check-proof test/regress/regress0/bv/core/slice-15.smt
unsat
```

**Comment by 4tXJ7f** [comment]
I've looked into this a bit and it looks like we do some rewriting with proofs that we don't do when unsat cores are enabled.

With unsat cores:
```
converting to CNF...
+ (not (= (_ bv65535 16) x))
+ (= (_ bv1 1) ((_ extract 15 15) x))
+ (= ((_ extract 14 0) x) ((_ extract 15 1) x))
+ true
solving...
```

Without unsat cores:
```
converting to CNF...
+ true
+ (= (concat (_ bv1 1) ((_ extract 14 1) BVSKOLEM$$_1)) BVSKOLEM$$_1)
+ (not (= (concat (_ bv1 1) BVSKOLEM$$_1) (_ bv65535 16)))
+ true
solving...
```

**Comment by yoni206** [comment]
@4tXJ7f can you check if this is already solved?

**Comment by 4tXJ7f** [comment]
I can confirm that this is not an issue anymore. I think it was fixed in #1957 (I mentioned the wrong issue number there).
